test/parallel/test-http-client-timeout-option-listeners.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+
+const agent = new http.Agent({ keepAlive: true });
+
+const server = http.createServer((req, res) => {
+  res.end('');
+});
+
+const options = {
+  agent,
+  method: 'GET',
+  port: undefined,
+  host: common.localhostIPv4,
+  path: '/',
+  timeout: common.platformTimeout(100)
+};
+
+server.listen(0, options.host, () => {
```

Comment:
It doesn't have to be on all callbacks. But, if the callback contains assertions, then we need to make sure they actually run.